src/index.ts

```diff
@@ -39,6 +40,81 @@ const {grpc} = new gax.GrpcClient(opts);
  */
 const PROJECT_ID_PLACEHOLDER = '{{projectId}}';
 
+// tslint:disable-next-line:no-any
+export type Metadata = any;
+
+/**
+ * @typedef {array} CreateTopicResponse
+ * @property {Topic} 0 The new {@link Topic}.
+ * @property {object} 1 The full API response.
+ */
+export type CreateTopicResponse = [Topic, object];
+
+/**
+ * @callback CreateTopicCallback
+ * @param {?Error} err Request error, if any.
+ * @param {Topic} topic The new {@link Topic}.
+ * @param {object} apiResponse The full API response.
+ */
+export interface CreateTopicCallback {
+  (err?: Error|null, topic?: Topic|null, apiResponse?: object): void;
+}
+
+export type Client = 'PublisherClient'|'SubscriberClient';
+
+export interface RequestConfig {
+  client: Client;
+  method: string;
+  reqOpts?: object;
+  gaxOpts?: CallOptions;
+}
+
+export interface GetClientConfig {
+  client: Client;
+  method?: string;
+}
+
+export interface RequestCallback<TResponse> {
+  (err?: Error|null, res?: TResponse|null): void;
+}
+
+/**
+ * @typedef {array} CreateSubscriptionResponse
+ * @property {Subscription} 0 The new {@link Subscription}.
+ * @property {object} 1 The full API response.
+ */
+export type CreateSubscriptionResponse = [Subscription, object];
+
+/**
+ * @callback CreateSubscriptionCallback
+ * @param {?Error} err Request error, if any.
+ * @param {Subscription} subscription The new {@link Subscription}.
+ * @param {object} apiResponse The full API response.
+ */
+export interface CreateSubscriptionCallback {
+  (err?: Error|null, subscription?: Subscription|null,
+   apiResponse?: object): void;
+}
+
+export interface CreateSubscriptionOptions extends SubscriptionMetadata {
+  flowControl?: {maxBytes?: number; maxMessages?: number;};
+  gaxOpts?: CallOptions;
+}
+
+/**
+ * Callback function to PubSub.getClient_().
+ * @internal
+ */
+interface GetClientCallback {
+  /**
+   * @param err - Error, if any.
+   * @param gaxClient - The gax client specified in RequestConfig.client.
+   *                    Typed any since it's importing Javascript source.
+   */
+  // tslint:disable-next-line:no-any
+  (err: Error|null, gaxClient?: any): void;
```

Comment:
There should be a good definition for the gax client, I just can't remember which. Gax is pretty well typed.

Reply:
[`gax.ClientStub`](https://github.com/googleapis/gax-nodejs/blob/master/src/grpc.ts#L107)